username_api.py

```diff
@@ -16,15 +17,37 @@ def check_username(website, username):
 	'behance'   :'https://{}.net/{}'.format(website, username)
 	}.get(website, 'https://{}.com/{}'.format(website, username)) # default
 
-	if website in ['pinterest', 'gitlab']:
-		res  = r.get(url)
-		code = 200 if bytes(username, encoding='utf-8') in res.content \
-			else 404
+	if check_format(website, username):
+		if website in ['pinterest', 'gitlab']:
+			res  = r.get(url)
+			code = 200 if bytes(username, encoding='utf-8') in res.content \
+				else 404
 
-		return {'status': code, 'url': url}
-	
+			return {'status': code, 'url': url}
+
+		else:
+			return {'status': r.get(url).status_code, 'url': url}
+	else:
+		return {'status': 400, 'url': url}
+
+def check_format(website, username):
+	pattern = {
+	'pinterest'  :r'^[a-zA-Z0-9_]{3,15}$',
```

Comment:
bad indent level.  And IMO this is data, and should be in a JSON file or python module.

Eventually it should be submitted to https://github.com/manu-chroma/username-availability-checker/issues/11#issuecomment-352158545 , and we use their definitions which is now https://github.com/manu-chroma/username-availability-checker/issues/25

Reply:
I kept indentation similar to the indentation on the existing lines https://github.com/manu-chroma/username-availability-checker/blob/master/username_api.py#L12-L17 – should fix this one and keep the other one how it is?